exchange/exchange.go

```diff
@@ -936,6 +936,88 @@ func (e *exchange) getBidCacheInfo(bid *pbsOrtbBid, auction *auction) (cacheInfo
 	return
 }
 
+// getAuctionCurrencyRates returns the constant currency conversion rates that will be used to process this
+// auction. Parameter customRates contains, if any, the bidRequest custom currency conversion rates
+// defined in bidRequest.Ext.  If usePbsRates flag is set to false, all previous Conversions values are discarded
+// and this function merely substitutes Conversions with customRates. But if usePbsRates was ommitted or set to
+// true, customRates' values update or add to the Conversions map
+func (e *exchange) getAuctionCurrencyRates(customRates *openrtb_ext.ExtRequestCurrency) currency.Conversions {
+	// No bidRequest.ext custom rates defined, use PBS rates as usual
+	if customRates == nil || len(customRates.ConversionRates) == 0 {
+		return e.currencyConverter.Rates()
+	}
+
+	// bidRequest.ext defines custom rates of its own, figure out whether to use a mix of
+	// Pbs rates and custom rates or custom rates only
+	if customRates.UsePBSRates != nil && !(*customRates.UsePBSRates) {
+		// Use custom rates only
+		includeInverseRates(customRates.ConversionRates)
```

Comment:
In case when only custom rates are in play, I believe we don't need to store the inverse rate conversions as well. The `GetRate` function in `rates.go` should take care of it when it does the currency conversion between two currencies

Reply:
True. Good catch, corrected.